docs/06_how-to-guides/40_multi-index/how-to-define-a-singleton.md

```diff
@@ -2,56 +2,90 @@
-```diff
-// singleton contract constructor
-singleton_example( name receiver, name code, datastream<const char*> ds ) :
-   contract(receiver, code, ds),
-+  singleton_instance(receiver, receiver.value)
-   { }
-}
-```
+Make sure you have the following prerequisites in place:
+
+* An EOSIO development environment, for details consult the [Get Started Guide](https://developers.eos.io/welcome/latest/getting-started-guide/index),
+
+## Procedure
+
+A singleton uses a single multi-index table to store named objects of various types. To define a simple singleton, which is storing an account `name` as primary value and a `uint64_t` as secondary value in structure `testtable`, follow the steps below:
+
+### 1. Preparation And Initialization
+
+Include the `eosio.hpp` and `singleton.hpp` headers and declare the `eosio` namespace usage
+
+   ```cpp
+   #include <eosio/eosio.hpp>
+   #include <eosio/singleton.hpp>
+   using namespace eosio;
+   ```
 
-Now you have defined and instantiated a singleton. Below you can find a possible implementation for the full class singleton example contract.
+### 2. Define The Table Data Structure
+
+Define the data structure for the multi-index table the singleton will use:
+
+   ```cpp
+   struct [[eosio::table]] testtable {
+      name primary_value;
+      uint64_t secondary_value;
+   };
+   ```
+
+### 3. Define A Singleton Type Alias
+
+For ease of use, define a type alias `singleton_type` based on the `eosio::singleton` template type, parametarized with a random name `"testtable"`, which has to respect the EOSIO account name restrictions, and the `testtable` data structure defined above
+
+   ```diff
+   struct [[eosio::table]] testtable {
+      name primary_value;
+      uint64_t secondary_value;
+   };
+   +using singleton_type = eosio::singleton<"testtable"_n, testtable>;
+   ```
+
+### 4. Define The Singleton Instance
+
+Define the singleton table instance as a data member of type `singleton_type` defined in the previous step.
+
+   ```diff
+   struct [[eosio::table]] testtable {
+      name primary_value;
+      uint64_t secondary_value;
+   };
+
+   using singleton_type = eosio::singleton<"testtable"_n, testtable>;
+   +singleton_type singleton_instance;
+   ```
+
+### 5. Initialize And Use The Singleton Instance
+
+Initialize `singleton_instance` defined previously by passing to its constructor the `receiver` and the `code` (in this case `receiver.value`) parameters; these two combined with "testtable" provide access to the partition of the RAM cache used by this singleton. In this example you will initialize the `singleton_instance` data member in the smart contract constructor, see below:
```

Comment:
no need for  "defined previously"

Reply:
Initialize the `singleton_instance` using the  constructor with the parameters `receiver` and `code`.  these parameters, combined with "testtable", provide access to the partition of the RAM cache used by this singleton. In our example you initialize the `singleton_instance` data member in the smart contract constructor, see below: